PSPs/drafts/psp-32.md

```diff
@@ -0,0 +1,837 @@
+	fn set_asset_attribute(
+		owner: Origin, 
+		asset: T::AssetId, 
+		key: T::AttributeKey,
+		value: T::AttributeValue);
+
+	/// Removes the `key` attribute from the given `asset`.
+	fn clear_asset_attribute( owner: Origin, asset: T::AssetId, key: T::AttributeKey);
+
+	/// Returns the token attribute of `key` for  `token` of `asset` .
+	fn token_attribute( 
+		asset: T::AssetId, 
+		token: T::TokenId, 
+		key: T::AttributeKey
+	) -> Option<T::AttributeValue>;
+
+	/// Updates attribute `key` of the `token` of `asset` using the given `value`.
+	fn set_token_attribute(
+		owner: Origin,
+		asset: T::AssetId,
+		token: T::TokenId,
+		key: T::AttributeKey,
+		value: T::AttributeValue);
+
+	/// Removes attribute `key` of the `token` of `asset`.
+	fn clear_token_attribute(
+		owner: Origin, 
+		asset: T::AssetId,
+		token: T::TokenId,
+		key: T::AttributeKey);
+}
+```
+
+### Common Types
+
+```Rust
+/// Extension ID.
+pub type ExtensionId = u16;
+
+/// Bounded-sized for attribute keys.
+pub type AttributeKey = BoundedVec<u8, KeyLimit>;
+
+/// Bounded-sized for attribute values.
+pub type AttributeValue = BoundedVec<u8, ValueLimit>;
+```
+
+### Extension Detection
+
+This standard **can be extended** with future proposals or optional capabilities. They _are optional_,
+so some networks may not implement them.
+
+The **extension_version** function is required to check if a given asset (in its current
+network) supports specific capabilities.
+
+The `ExtensionId` type implementation **requires** the following statements:
+- It should be **flexible enough to accept new future values** without a serialization error. It
+  means that new extension IDs can be used to query legacy assets and no error should be raised. In
+  that case, `extension_version` will return `None`.
+- It should be enough to **store up to 2^16 extension identifiers**.
+
+On `ETH`, [ERC-165](https://eips.ethereum.org/EIPS/eip-165) could be used to support this detection.
+
+See [`Paratoken::extension_version(..)`](#base-interface).
+
+```Rust
+
+/// Defines a [Semantic Versioning](https://semver.org/)
+pub struct ExtensionVersion {
+	/// The **major version** is incremented when an incompatible API change is done.
+	pub major: u64,
+	/// The **minor version** is incremented when a new functionality in a backwards compatible 
+	/// manner is added.
+	pub minor: u64,
+	/// The **patch version** is incremented when a backwards compatible bug is fixed.
+	pub patch: u64,
+}`
+```
+
+#### Registered Extensions
+
+The following table contains the extensions already registered by this standard.
+
+| Ext. ID | Version |  Extension |
+| ------: | :-----: | :--------- |
+| 0x0001  | 1.0.0   | [Batch Extension](extensions/batch.md) |
+| 0x0002  | 1.0.0   | [Chunks Extension](extensions/chunks.md) |
+| 0x0003  | 1.0.0   | [Enumerable Extension](extensions/enumerable.md) |
+| 0x0004  | 0.1.0   | **TBD** Teleportation |
+| 0x0005  | 1.0.0   | [Operator](extensions/operator.md) |
```

Comment:
I can't find `extensions/*`

Reply:
after scrolling down I found the `Extension: *` sections of `psp-32.md`
wouldn't it make more sense to point at these sections, instead of `extensions/*`?